src/meta/client/MetaClient.cpp

```diff
@@ -1622,6 +1622,20 @@ const std::vector<HostAddr>& MetaClient::getAddresses() {
     return addrs_;
 }
 
+StatusOr<std::vector<nebula::cpp2::RoleItem>>
+MetaClient::getRolesByUser(const std::string& user) {
```

Comment:
We should hold a cache for User->Role map

Reply:
> We should hold a cache for User->Role map

Very good advice. changed the role map to cache.